### DISCUSSION TITLE
Can someone please help me to install erlang through homebrew Apple M1 chip

### DISCUSSION THREAD

**Discussion by mdsardar** [Homebrew/discussions#872] [Getting started]
Greetings to All,

I'm new to mac and was learning about Homebrew and installation of erlang through homebrew.  I could see that brew has been installed when I'm trying to install erlang getting the below response. The system has an M1 chip. 

> Homebrew 3.0.1
> Homebrew/homebrew-core (no Git repository)
> Homebrew/homebrew-cask (git revision ddca2; last commit 2021-02-20)
> 

While installing homebrew getting the below response. 

**> brew install erlang
> Updating Homebrew...
> fatal: Could not resolve HEAD to a revision
> ==> Searching for similarly named formulae...
> Error: No similarly named formulae found.
> Error: No available formula or cask with the name "erlang".**

I had read the brew supported formula through `https://formulae.brew.sh/formula/erlang`, can see erlang is supported, but sI couldn't crack down on the issue, can someone help me to install erlang?

Thanks & Best

**Comment by gromgit** [comment]
What's the output of `brew config` and `brew doctor`?

**Reply by mdsardar** [reply]
$ brew config
HOMEBREW_VERSION: 3.0.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: c951be8d3c7b339c7e759b7c40aec859e09a70a5
Last commit: 9 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: HEAD
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.0.1-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
Rosetta 2: false

```
```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
```

**Reply by carlocab** [reply]
Have you tried the fix suggested by `brew doctor`? This one:
```
git -C $(brew --repo homebrew/core) checkout master
```

**Reply by mdsardar** [reply]
Ok, the brew doctor suggests some treatment. Now I see and tried the same on my current directory and getting as already on master 

> $ git -C $(brew --repo homebrew/core) checkout master
> Branch 'master' set up to track remote branch 'master' from 'origin'.
> Already on 'master'
> 

Before I tried with brew update command and got to know brew is already up-to-date.

**Reply by mdsardar** [reply]
Just now, I tried to update brew again and got erl lang installed successfully !! :)

**Reply by zerowolrd** [reply]
> Have you tried the fix suggested by `brew doctor`? This one:
> 
> ```
> git -C $(brew --repo homebrew/core) checkout master
> ```

Thank you man :)

**Comment by mdsardar** [comment]
Thank you to All who helped me...

**Comment by sailor119** [comment]
I had met the same problem, and according to your example case, also fixed the issue. Thanks All!
